main.py

```diff
@@ -4,13 +4,22 @@
 import urllib.request as request
 import sys
 
+regex_doi = r"doi\s?=\s?[{\"']([^}\"']+)[}\"']"
+regex_url = r"url\s?=\s?[{\"'](https?://doi\.org/([\w\.]+/[^}/\"']+))/?[}\"']"
+
 
 for line in fileinput.input():
     try:
-        if "doi={" in re.sub("\s+", "", line).lower():
-            doi = re.search("{(.+)}", line).group(1)
+        search_doi = re.search(regex_doi, re.sub("\s+", "", line).lower(), re.IGNORECASE)
+        search_url = re.search(regex_url, re.sub("\s+", "", line).lower(), re.IGNORECASE)
+        doi = (
+            (search_doi and search_doi.group(1)) or
+            (search_url and search_url.group(2))
+        )
+
+        if doi:
             data = json.load(request.urlopen("http://shortdoi.org/%s?format=json" % doi))
-            print(line.replace(doi, data["ShortDOI"]), end="")
+            print(line.replace(doi, data["ShortDOI"][3 * bool(search_url):]), end="")
```

Comment:
What does `[3 * bool(search_url):]` do here?

Reply:
Apparently the URL for a shortened doi (e.g., `10/f6xchr `) can be shortened even more by leaving out `10/`. `3 * bool(search_url)` simply says skip the first 3 chars if the shortened doi is coming from an URL. So this ensures that `https://doi.org/f6xchr` instead of `https://doi.org/10/f6xchr` is printed.

Reply:
hmm, I found that if you omit the '10/', some things break. For example https://doi2bib.org/. Could you undo this change?

Reply:
Ohh, I see. You only do this for URLs. How about we add a new variable that is

```python
short_doi = data["shortDOI"][3:] if search_url else data["shortDOI"]
```

Reply:
It makes it a bit more obvious what's happening.

Reply:
Done :)